aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -298,8 +298,13 @@ private List<String> deleteBatch(String bucket, Collection<String> keysToDelete)
   @Override
   public Iterable<FileInfo> listPrefix(String prefix) {
     S3URI s3uri = new S3URI(prefix, s3FileIOProperties.bucketToAccessPointMapping());
+    String suffix = "";
+    if (S3Express.checkIfS3Express(s3uri.bucket()) && !prefix.endsWith("/")) {
+      suffix = "/";
+    }
+
     ListObjectsV2Request request =
-        ListObjectsV2Request.builder().bucket(s3uri.bucket()).prefix(s3uri.key()).build();
+        ListObjectsV2Request.builder().bucket(s3uri.bucket()).prefix(s3uri.key() + suffix).build();
```

Comment:
help me understand, what would be the behavior if I list objects without a `/` in a directory bucket? would it fail, or list assuming it is a directory?

Reply:
It throws an unsupported exception.
https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_RequestSyntax under prefix you can see the docs on how it's not supported